src/_pytest/python.py

```diff
@@ -1343,9 +1344,24 @@ def _idval(
         # Name of a class, function, module, etc.
         name: str = getattr(val, "__name__")
         return name
+    # TODO: Remove type-ignore after next mypy release.
+    # https://github.com/python/typeshed/commit/076983eec45e739c68551cb6119fd7d85fd4afa9
+    elif isinstance(val, os.PathLike):  # type: ignore[misc]
+        return _idval_pathlike(val, config)
+
     return str(argname) + str(idx)
 
 
+def _idval_pathlike(pathlike: "os.PathLike[AnyStr]", config: Optional[Config]) -> str:
+    # For consistency, always use Unix path separators.
+    path: Union[str, bytes] = os.fspath(pathlike)
```

Comment:
The type annotation here should not be necessary (it should be automatically inferred)